### PR TITLE
CRYPTO-45: Document how to build Commons Crypto

### PR DESCRIPTION
**Pull request by dianfu**
This PR adds the Developer Guide. For User Guide, it's already available in src/site/xdoc/userguide.xml.

### CODE REVIEW

File: BUILDING.txt [hunk 1]
```diff
@@ -0,0 +1,74 @@
+Build instructions for Apache Commons Crypto
+
+----------------------------------------------------------------------------------
+Requirements:
+
+* Unix System
+* JDK 1.6+ (environment variable JAVA_HOME must be set)
+* Java Cryptography Extension (JCE) Unlimited Strength Jurisdiction Policy
+  Files (if running unit tests)
+* Maven 3.0 or later
```
Comment (winningsix) [inline BUILDING.txt]: Better to be "or above"

File: BUILDING.txt [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+* Make
+* Openssl devel 1.0.1c+
```
Comment (winningsix) [inline BUILDING.txt]: Maybe we could follow naming convention as Maven part.

File: pom.xml [hunk 3]
```diff
@@ -331,6 +331,15 @@ Features
         <groupId>org.apache.rat</groupId>
         <artifactId>apache-rat-plugin</artifactId>
         <version>${commons.rat.version}</version>
+        <configuration>
+          <excludes>
+          <exclude>.gitattributes</exclude>
```
Comment (sundapeng) [inline pom.xml]: add two blanks.

**Comment by dianfu** [comment]
@winningsix and @sundapeng, thanks for your review. Updated the PR.

File: BUILDING.txt [hunk 4]
```diff
@@ -0,0 +1,74 @@
+Build instructions for Apache Commons Crypto
```
Comment (jerrychenhf) [inline BUILDING.txt]: Please use "OpenSSL" as the formal name in this document.

File: BUILDING.txt [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+For JDK 1.6: http://www.oracle.com/technetwork/java/javase/downloads/jce-6-download-429243.html
+For JDK 1.7: http://www.oracle.com/technetwork/java/javase/downloads/jce-7-download-432124.html
+For JDK 1.8: http://www.oracle.com/technetwork/java/javase/downloads/jce8-download-2133166.html
+
+Install JCE Unlimited Strength Jurisdiction Policy Files to JDK:
+
+Copy downloaded local_policy.jar and US_export_policy.jar to <java-home>/jre/lib/security/
+
+----------------------------------------------------------------------------------
+Upgrade Openssl version in Mac:
```
Comment (jerrychenhf) [inline BUILDING.txt]: Do we need name this section to Verify OpenSSL version?
My idea is:
check OpenSSL version is common for all OS.

upgrade OpenSSL version is different for different OS, we can have a small section as:
Upgrade Openssl in Mac:
...

For Linux systems, there are a lot of distributions, we can simply mention:
Upgrade OpenSSL in Linux:
You can follow your OS distribution instructions to upgrade OpenSSL to a proper version.

File: BUILDING.txt [hunk 6]
```diff
@@ -0,0 +1,74 @@
+Build instructions for Apache Commons Crypto
+
+----------------------------------------------------------------------------------
+Requirements:
+
+* Unix System
```
Comment (jerrychenhf) [inline BUILDING.txt]: Can we use:
Unix System (Linux or Mac)

**Comment by jerrychenhf** [comment]
Thanks @dianfu for contribution. I left a few comments.

**Comment by dianfu** [comment]
Thanks @jerrychenhf for your review. Updated the patch.

**Pull request closed without merging**